crates/forge-runner/src/build_trace_data.rs

```diff
@@ -35,18 +36,39 @@ pub const TEST_CODE_FUNCTION_NAME: &str = "SNFORGE_TEST_CODE_FUNCTION";
 pub fn build_profiler_call_trace(
     value: &Rc<RefCell<CallTrace>>,
     contracts_data: &ContractsData,
+    test_artifacts_path: &Utf8PathBuf,
 ) -> ProfilerCallTrace {
     let value = value.borrow();
 
+    let entry_point = build_profiler_call_entry_point(value.entry_point.clone(), contracts_data);
     let vm_trace = value.vm_trace.as_ref().map(|trace_data| {
         trace_data
             .iter()
             .map(build_profiler_trace_entry)
             .collect_vec()
     });
 
+    let cairo_execution_info = if vm_trace.is_some() && entry_point.contract_name.is_some() {
+        let contract_name = entry_point.contract_name.as_ref().unwrap();
+        let source_sierra_path = if contract_name == TEST_CODE_CONTRACT_NAME {
+            test_artifacts_path
+        } else {
+            contracts_data
+                .get_source_sierra_path(contract_name)
+                .unwrap()
+        }
+        .clone();
```

Comment:
I have no idea what this large if statement does. Please provide some abstraction.

Reply:
Done